api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -187,11 +192,28 @@ public void validateInstitutionalAffiliation(Profile profile) {
    * @param previousProfile
    */
   public void updateProfileForUser(DbUser user, Profile updatedProfile, Profile previousProfile) {
+    updateProfileForUser(user, updatedProfile, previousProfile, false);
+  }
+
+  /**
+   * Updates a profile for a given user as Admin and persists all information to the database.
+   *
+   * @param user
+   * @param updatedProfile
+   * @param previousProfile
+   */
+  public void adminUpdateProfileForUser(
+      DbUser user, Profile updatedProfile, Profile previousProfile) {
+    updateProfileForUser(user, updatedProfile, previousProfile, true);
+  }
+
+  private void updateProfileForUser(
+      DbUser user, Profile updatedProfile, Profile previousProfile, boolean userIsAdmin) {
```

Comment:
I'm not super happy about the boolean flag here - open to ideas

Reply:
That seems reasonable.  One snag here is that sometimes the user object is "without authorities" and I don't understand the implications of that well enough

Reply:
Aren't they just lazily fetched the first time you try to grab them?

```
  // Authorities (special permissions) are granted using api/project.rb set-authority.
  @ElementCollection(fetch = FetchType.LAZY)
  @CollectionTable(name = "authority", joinColumns = @JoinColumn(name = "user_id"))
  @Column(name = "authority")
  public Set<Short> getAuthorities() {
    return authorities;
  }
```

This is getting at my main source of confusion: does the DAO or the Entity drive the train here?

Reply:
Oh I see, thanks.  This seems much more tractable, then.

Reply:
This logic is now gated on the new DbUser.hasAuthority()